google-cloud-firestore/src/main/java/com/google/cloud/firestore/Firestore.java

```diff
@@ -85,6 +85,31 @@
   @Nonnull
   <T> ApiFuture<T> runTransaction(@Nonnull final Transaction.Function<T> updateFunction);
 
+  /**
+   * Executes the given updateFunction and then attempts to commit the changes applied within the
+   * transaction. If any document read within the transaction has changed, the updateFunction will
+   * be retried. If it fails to commit after 5 attempts, the transaction will fail.
+   *
+   * @param updateFunction The function to execute within the transaction context.
```

Comment:
One potential downside of your proposal is that it makes it now much easier to invoke transactions that take a long time to execute (as the user may now be inclined to kick off other tasks as well). Can you add something to the comment to explain these downsides? E.g. something like: "Running a transaction places locks all all consumed documents. To unblock other clients, the Firestore backend automatically releases all locks after 60 seconds of inactivity and fails all transactions that last longer than 270 seconds (see https://firebase.google.com/docs/firestore/quotas#writes_and_transactions)"

Reply:
comment added to the javadoc